File: Summary Ranges - Leetcode 228/Summary Ranges - Leetcode 228.cpp
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ans;
        int n = nums.size();
        int i = 0;

        while (i < n) {
            int start = nums[i];
            while (i < n - 1 && nums[i] + 1 == nums[i + 1]) {
                i++;
            }
            if (start != nums[i]) {
                ans.push_back(to_string(start) + "->" + to_string(nums[i]));
            } else {
                ans.push_back(to_string(nums[i]));
            }
            i++;
        }

        return ans;
    }
};

File: Summary Ranges - Leetcode 228/Summary Ranges - Leetcode 228.js
var summaryRanges = function(nums) {
    let ans = [];
    let i = 0;

    while (i < nums.length) {
        let start = nums[i];
        while (i < nums.length - 1 && nums[i] + 1 === nums[i + 1]) {
            i++;
        }
        if (start !== nums[i]) {
            ans.push(start + "->" + nums[i]);
        } else {
            ans.push(start.toString());
        }
        i++;
    }

    return ans;
};
